command/agentproxyshared/cache/cachememdb/index.go

```diff
@@ -76,12 +77,17 @@ type Index struct {
 	// LastRenewed is the timestamp of last renewal
 	LastRenewed time.Time
 
-	// Type is the index type (token, auth-lease, secret-lease)
+	// Type is the index type (token, auth-lease, secret-lease, static-secret, token-capabilities)
 	Type string
 
+	// Capabilities is a set of known capabilities for the given token. Used only for
+	// token-capabilities type cache entries.
+	// Implemented as a map for uniqueness.
+	Capabilities map[string]struct{}
```

Comment:
This doesn't seem quite right: we have a map of tokens associated with this index entry, but Capabilities is only holding a flat set of capabilities.  Wouldn't it make more sense to make Tokens map from token to capabilities?

Do we even need the list of capabilities?  Don't we just care about `read`?

Reply:
Oh, I see now: these are actually a different kind of cache entry.  Why overload the Index type?  It seems more natural (and memory efficient) to me to create a new index in the cache table, and to store a new struct type in that index that's specific to capabilities, considering how few fields we actually need and how little overlap there is.

Reply:
Yeah, I did consider this approach. There are some shared fields, but ultimately there's already quite a lot of variation in what we store for different kinds of secrets already. To use an example: `Lease` isn't used by static secrets, nor is it even used by all dynamic secrets.

From what I could tell, keeping things together simplified the code quite a lot, and made aspects like persistence easier. I'm happy to revisit if you feel strongly that this shouldn't be this way, though.

Reply:
I do feel somewhat strongly, yes.  At least both static and dynamic secrets are conceptually related and have some overlap, though we may want to split them up too in order to minimize memory footprint - here I'm thinking particularly of the overhead of maintaining so many memdb indexes that aren't used in the static case, e.g. IndexNameRequestPath, IndexNameLeaseToken, etc.

Token capabilities are a completely different beast though and I don't think they should be treated in the same way as cached requests, even if it does make the code simpler right now.

Reply:
That makes sense! Let me have a crack at refactoring.

Reply:
Updated! It's a lot cleaner, now. I probably should have done this initially, but better late than never. Thanks!